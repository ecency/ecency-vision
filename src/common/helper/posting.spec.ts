import {
  createPermlink,
  extractMetaData,
  makeJsonMetaData,
  makeCommentOptions,
  makeJsonMetaDataReply,
  createReplyPermlink,
  createPatch,
} from './posting';

describe('createPermlink', () => {
  it('(1) ', () => {
    const input = 'lorem ipsum dolor sit amet';
    expect(createPermlink(input)).toMatchSnapshot();
  });
});

describe('createPermlink random', () => {
  jest.spyOn(Math, 'random').mockImplementation(() => {
    return 1.95136022969379;
  });

  it('(1) ', () => {
    const input = 'lorem ipsum dolor sit amet';
    expect(createPermlink(input, true)).toMatchSnapshot();
  });
});

describe('createPermlink non-latin chars', () => {
  jest.spyOn(Math, 'random').mockImplementation(() => {
    return 1.95136022969379;
  });

  it('(1) ', () => {
    const input = 'ปลาตัวใหญ่สีเหลืองทอง';
    expect(createPermlink(input)).toMatchSnapshot();
  });
});

describe('extractMetadata', () => {
  it('(1) ', () => {
    const input = '<img src="http://www.xx.com/a.png"> @lorem @ipsum';
    expect(extractMetaData(input)).toMatchSnapshot();
  });

  it('(2) ', () => {
    const input =
      '@lorem <img src="http://www.xx.com/a.png"> ![h74zrad2fh.jpg](https://img.esteem.ws/h74zrad2fh.jpg) http://www.google.com/foo/bar  @ipsum';
    expect(extractMetaData(input)).toMatchSnapshot();
  });
});

describe('makeJsonMetaData', () => {
  it('(1) ', () => {
    const meta = {
      image: [
        'http://www.xx.com/a.png',
        'https://img.esteem.ws/h74zrad2fh.jpg',
      ],
      links: ['http://www.google.com/foo/bar'],
      users: ['lorem', 'ipsum'],
    };
    const tags = ['esteem', 'art'];

    expect(makeJsonMetaData(meta, tags, '2.0.0')).toMatchSnapshot();
  });
});

describe('makeCommentOptions', () => {
  it('(1) Default 50% / 50%', () => {
    expect(
      makeCommentOptions('talhasch', 'lorem-ipsum-1', 'default'),
    ).toMatchSnapshot();
  });

  it('(2) Power Up 100%', () => {
    expect(
      makeCommentOptions('talhasch', 'lorem-ipsum-1', 'sp'),
    ).toMatchSnapshot();
  });

  it('(3) Decline Payout', () => {
    expect(
      makeCommentOptions('talhasch', 'lorem-ipsum-1', 'dp'),
    ).toMatchSnapshot();
  });

  it('(4) Empty beneficiary list', () => {
    expect(
      makeCommentOptions('talhasch', 'lorem-ipsum-1', 'default', []),
    ).toMatchSnapshot();
  });

  it('(5) With beneficiary list', () => {
    expect(
      makeCommentOptions('talhasch', 'lorem-ipsum-1', 'default', [
        {account: 'foo', weight: 300},
        {account: 'bar', weight: 200},
      ]),
    ).toMatchSnapshot();
  });
});

describe('makeJsonMetadataReply', () => {
  it('(1)', () => {
    expect(makeJsonMetaDataReply(['foo', 'bar'], '1.1')).toMatchSnapshot();
  });
});

describe('createReplyPermlink', () => {
  // @ts-ignore
  jest.spyOn(Date, 'now').mockImplementation(() => {
    return new Date('2018-09-21T12:00:50.000Z');
  });

  it('(1) ', () => {
    expect(createReplyPermlink('good-karma')).toMatchSnapshot();
  });
});

describe('createPatch', () => {
  it('(1) ', () => {
    expect(
      createPatch('lorem ipsum dlor sit amet', 'lorem ipsum dolor sit amet'),
    ).toMatchSnapshot();
  });
});
