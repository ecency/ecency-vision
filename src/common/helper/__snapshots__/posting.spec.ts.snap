// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createPatch (1)  1`] = `
"@@ -6,16 +6,17 @@
  ipsum d
+o
 lor sit 
"
`;

exports[`createPermlink (1)  1`] = `"lorem-ipsum-dolor-sit-amet"`;

exports[`createPermlink non-latin chars (1)  1`] = `"f38c58113ff7e"`;

exports[`createPermlink random (1)  1`] = `"lorem-ipsum-dolor-sit-amet-f38c58113ff7e"`;

exports[`createReplyPermlink (1)  1`] = `"re-good-karma-2018921t120500z"`;

exports[`extractMetadata (1)  1`] = `
Object {
  "image": Array [
    "http://www.xx.com/a.png",
  ],
  "thumbnails": Array [
    "http://www.xx.com/a.png",
  ],
  "users": Array [
    "lorem",
    "ipsum",
  ],
}
`;

exports[`extractMetadata (2)  1`] = `
Object {
  "image": Array [
    "http://www.xx.com/a.png",
    "https://img.esteem.ws/h74zrad2fh.jpg",
  ],
  "links": Array [
    "http://www.google.com/foo/bar",
  ],
  "thumbnails": Array [
    "http://www.xx.com/a.png",
    "https://img.esteem.ws/h74zrad2fh.jpg",
  ],
  "users": Array [
    "lorem",
    "ipsum",
  ],
}
`;

exports[`makeCommentOptions (1) Default 50% / 50% 1`] = `
Object {
  "allow_curation_rewards": true,
  "allow_votes": true,
  "author": "talhasch",
  "extensions": Array [],
  "max_accepted_payout": "1000000.000 HBD",
  "percent_hbd": 10000,
  "permlink": "lorem-ipsum-1",
}
`;

exports[`makeCommentOptions (2) Power Up 100% 1`] = `
Object {
  "allow_curation_rewards": true,
  "allow_votes": true,
  "author": "talhasch",
  "extensions": Array [],
  "max_accepted_payout": "1000000.000 HBD",
  "percent_hbd": 0,
  "permlink": "lorem-ipsum-1",
}
`;

exports[`makeCommentOptions (3) Decline Payout 1`] = `
Object {
  "allow_curation_rewards": true,
  "allow_votes": true,
  "author": "talhasch",
  "extensions": Array [],
  "max_accepted_payout": "0.000 HBD",
  "percent_hbd": 10000,
  "permlink": "lorem-ipsum-1",
}
`;

exports[`makeCommentOptions (4) Empty beneficiary list 1`] = `
Object {
  "allow_curation_rewards": true,
  "allow_votes": true,
  "author": "talhasch",
  "extensions": Array [],
  "max_accepted_payout": "1000000.000 HBD",
  "percent_hbd": 10000,
  "permlink": "lorem-ipsum-1",
}
`;

exports[`makeCommentOptions (5) With beneficiary list 1`] = `
Object {
  "allow_curation_rewards": true,
  "allow_votes": true,
  "author": "talhasch",
  "extensions": Array [
    Array [
      0,
      Object {
        "beneficiaries": Array [
          Object {
            "account": "foo",
            "weight": 300,
          },
          Object {
            "account": "bar",
            "weight": 200,
          },
        ],
      },
    ],
  ],
  "max_accepted_payout": "1000000.000 HBD",
  "percent_hbd": 10000,
  "permlink": "lorem-ipsum-1",
}
`;

exports[`makeJsonMetaData (1)  1`] = `
Object {
  "app": "proofofbrain-blog/2.0.0-vision",
  "description": "",
  "format": "markdown+html",
  "image": Array [
    "http://www.xx.com/a.png",
    "https://img.esteem.ws/h74zrad2fh.jpg",
  ],
  "links": Array [
    "http://www.google.com/foo/bar",
  ],
  "tags": Array [
    "esteem",
    "art",
  ],
  "users": Array [
    "lorem",
    "ipsum",
  ],
}
`;

exports[`makeJsonMetadataReply (1) 1`] = `
Object {
  "app": "proofofbrain-blog/1.1-vision",
  "format": "markdown+html",
  "tags": Array [
    "foo",
    "bar",
  ],
}
`;
